inc/notificationeventmailing.class.php

```diff
@@ -137,27 +137,38 @@ static public function send(array $data) {
          }
          $mmail->Subject  = $current->fields['name'];
 
+         // manage item attached documents
+         $document_items = $DB->request('glpi_documents_items', [
+            'items_id' => $current->fields['items_id'],
+            'itemtype' => $current->fields['itemtype'],
+         ]);
+
+         $doc = new Document();
+
          if (empty($current->fields['body_html'])) {
             $mmail->isHTML(false);
             $mmail->Body = $current->fields['body_text'];
+
+            if ($CFG_GLPI['attach_ticket_documents_to_mail']) {
+               // manage item attached documents
+               foreach ($document_items as $doc_i_data) {
+                  self::attachDocument($mmail, $doc, $doc_i_data);
+               }
+            }
          } else {
             $mmail->isHTML(true);
             $mmail->Body = '';
             $current->fields['body_html'] = Html::entity_decode_deep($current->fields['body_html']);
 
-            // manage item attached documents
-            $document_items = $DB->request('glpi_documents_items', [
-               'items_id' => $current->fields['items_id'],
-               'itemtype' => $current->fields['itemtype'],
-            ]);
             $inline_docs = [];
-            $doc = new Document();
             if (count($document_items)) {
                foreach ($document_items as $doc_i_data) {
-                  $doc->getFromDB($doc_i_data['documents_id']);
                   // Add embeded image if tag present in ticket content
                   if (preg_match_all('/'.Document::getImageTag($doc->fields['tag']).'/',
                                      $current->fields['body_html'], $matches, PREG_PATTERN_ORDER)) {
+
```

Comment:
This part should be moved above l167 (`if (preg_match_all('/'.Document::getImageTag($doc->fields['tag']).'/',`)

Reply:
What exactly shuld be removed? You mean to put cycle in attachDocuments and use lambda for inner logic?